apps/champions/lib/champions/battle/simulator.ex

```diff
@@ -621,6 +621,34 @@ defmodule Champions.Battle.Simulator do
     [caster.id]
   end
 
+  defp choose_targets_by_strategy(
+         caster,
+         %{type: %{"lowest" => stat}, target_allies: target_allies, count: count},
+         state
```

Comment:
I have a few suggestions regarding these 2 new function signatures:
1. I wouldn't extract map keys that will only be used in the body and not for pattern-match use. In this case, I would just match the "type" which is the field that differs in these functions. (See https://hexdocs.pm/elixir/main/code-anti-patterns.html#complex-extractions-in-clauses).
2. I don't like some functions extracting "type" field as a value and others as a nested map key. I think every function should do the same, being just the type a value for type key my preferable option. If you need an additional parameter (stat) just add a new key to the map.

Reply:
As discussed, we'll solve your second comment in a separate issue: https://github.com/lambdaclass/afk_gacha_game/issues/479